vm.go

```diff
@@ -2839,9 +2835,6 @@ func (numargs call) exec(vm *vm) {
 	obj := vm.toCallee(v)
 
 repeat:
-	if vm.r.stackDepthLimit != 0 && len(vm.callStack)+1 >= vm.r.stackDepthLimit {
```

Comment:
[nit]/[q] I'm not sure how much risk we remove this condition in this version of goja. If we want to remove `stackDepthLimit`, do you think we can keep this condition but use `vm.maxCallStackSize` as a guard here like the below?
```
if vm.maxCallStackSize != 0 && len(vm.callStack)+1 >= vm.maxCallStackSize {

Reply:
Added back `stackDepthLimit`